app/models/identity_header.rb

```diff
@@ -10,8 +10,14 @@ def initialize(b64_identity)
     @b64_identity = b64_identity
   end
 
+  def blank?
+    @b64_identity.blank?
+  end
+
   def valid?
     identity.present? && entitled?
+  rescue JSON::ParserError, NoMethodError
+    false
   end
```

Comment:
This might not work as you expect. I try to completely avoid returns from a method-level rescue because it's so confusing. Please add a `begin` block.

Reply:
Can you please elaborate on the unexpected behavior? The `begin` is implicit for the method block.

Reply:
We've just been bitten by this in the past, even within the authentication logic. I can't see why this won't work - I think we ran into issues in the past with explicit `return` statements because the order of execution is a bit strange:

```rb
def foo(r=false)
  raise 'no!' if r
  puts 'one'
  1
rescue
  puts 'two'
  2
ensure
  puts 'three'
  3
end

[11] pry(main)> foo
one
three
=> 1

[12] pry(main)> foo(true)
two
three
=> 2
```

I think we're good here :+1: Sorry for the concern.